orchestrator/careplancontributor/handle_taskfiller.go

```diff
@@ -37,9 +37,9 @@ func (s *Service) handleTaskFillerCreateOrUpdate(ctx context.Context, cpsClient
 	// If partOfRef is nil, handle the task as a primary task - no need to create follow-up subtasks for newly created Tasks
 	//This only happens on Task update where the Task.output is filled with a QuestionnaireResponse
 	if partOfRef == nil {
-		// Check if the priary task is "created", its status will be updated by subtasks that are completed - not directly here
-		if task.Status != fhir.TaskStatusRequested {
-			log.Debug().Msg("primary Task.status != requested (workflow already started) - not processing in handleTaskFillerCreateOrUpdate")
+		// Check if the primary task is "created", its status will be updated by subtasks that are completed - not directly here
+		if task.Status != fhir.TaskStatusRequested && task.Status != fhir.TaskStatusReceived {
```

Comment:
Considering we use this again lower down, do we maybe want to centralise this condition somewhere? Will make future updates easier as well if we want to exclude more statuses

Reply:
Doesn't this create a subtask twice when the primary Task goes from `requested` to `received`? Creating a new Task with the status `received` is not allowed:

<img src="https://santeonnl.github.io/shared-care-planning/Task-state-machine.png"/>

Reply:
> Considering we use this again lower down, do we maybe want to centralise this condition somewhere? Will make future updates easier as well if we want to exclude more statuses

Yes, and I think the task engine can use some cleanup/refactoring, as it has become quite the complex beast. But let's do that in a different PR.